### ISSUE TITLE
GEOPOS throws UnsupportedOperationException in 4.4.0.Final

### ISSUE THREAD

**Issue by lerouxrgd** [lettuce-io/lettuce-core#593]
Running unit tests on my project after upgrading to latest 4.4.0.Final I run into a `UnsupportedOperationException` when using `.geopos(...)`.
I got this error with sync and async commands, regular Redis and RedisCluster.
I am using Java 1.8.0_91.

The test is simply checking that a call to `.geopos(...)` works (following [official example](https://redis.io/commands/geopos#examples))
This used to work up to 4.3.3.Final version.
Since 4.4.0.Final it looks like `GeoCoordinatesListOutput`'s `set(ByteBuffer bytes)` now uses:
```java
subscriber.onNext(output, new GeoCoordinates(x, value));
```
However `output` is initialized with a `Collections.emptyList()` in the constructor:
```java
public GeoCoordinatesListOutput(RedisCodec<K, V> codec) {
        super(codec, Collections.emptyList());
        setSubscriber(ListSubscriber.instance());
}
```
Later on `subscriber` will try to call `.add(...)` on it, which I suspect to be the issue.

Here is the full stacktrace I get:
```java
Caused by: java.lang.UnsupportedOperationException: null
 at java.util.AbstractList.add (AbstractList.java:148)
    java.util.AbstractList.add (AbstractList.java:108)
    com.lambdaworks.redis.output.ListSubscriber.onNext (ListSubscriber.java:47)
    com.lambdaworks.redis.output.GeoCoordinatesListOutput.set (GeoCoordinatesListOutput.java:55)
    com.lambdaworks.redis.protocol.RedisStateMachine.safeSet (RedisStateMachine.java:429)
    com.lambdaworks.redis.protocol.RedisStateMachine.decode (RedisStateMachine.java:233)
    com.lambdaworks.redis.protocol.CommandHandler.decode (CommandHandler.java:313)
    com.lambdaworks.redis.protocol.CommandHandler.decode (CommandHandler.java:257)
    com.lambdaworks.redis.protocol.CommandHandler.channelRead (CommandHandler.java:241)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:362)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:348)
    io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:340)
    io.netty.channel.ChannelInboundHandlerAdapter.channelRead (ChannelInboundHandlerAdapter.java:86)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:362)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:348)
    io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:340)
    io.netty.channel.ChannelInboundHandlerAdapter.channelRead (ChannelInboundHandlerAdapter.java:86)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:362)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:348)
    io.netty.channel.AbstractChannelHandlerContext.fireChannelRead (AbstractChannelHandlerContext.java:340)
    io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead (DefaultChannelPipeline.java:1334)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:362)
    io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead (AbstractChannelHandlerContext.java:348)
    io.netty.channel.DefaultChannelPipeline.fireChannelRead (DefaultChannelPipeline.java:926)
    io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read (AbstractNioByteChannel.java:134)
    io.netty.channel.nio.NioEventLoop.processSelectedKey (NioEventLoop.java:644)
    io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized (NioEventLoop.java:579)
    io.netty.channel.nio.NioEventLoop.processSelectedKeys (NioEventLoop.java:496)
    io.netty.channel.nio.NioEventLoop.run (NioEventLoop.java:458)
    io.netty.util.concurrent.SingleThreadEventExecutor$5.run (SingleThreadEventExecutor.java:858)
    io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run (DefaultThreadFactory.java:138)
    java.lang.Thread.run (Thread.java:745)
```

**Comment by mp911de** [comment]
This issue duplicates #591. Can you give `4.4.1-SNAPSHOT` a try?

**Comment by lerouxrgd** [comment]
Yes it fixes it, sorry I didn't see #591 before.
I'll wait for 4.4.1 release then.
Thanks.
